File: Ruby/sorpresa_pagar_cobrar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'sorpresa'
module Civitas
  class Sorpresa_pagar_cobrar < Sorpresa
    attr_accessor :valor
    def initialize
      #Sorpresa.new()
      @valor = 10
    end

    def aplicar_a_jugador(actual, todos)
      if (jugador_correcto(actual, todos))
        informe(actual, todos)
        todos[actual].modificarSaldo(@valor)
      end
    end

    def to_s()
      "Pagar cobrar"
    end
    public_class_method:new
  end
end

File: Java/Sorpresa_PagarCobrar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ilzk
 */
public class Sorpresa_PagarCobrar extends Sorpresa {
    
    public Sorpresa_PagarCobrar(){
        valor = 10;
    }
    
    @Override
    /**Aplica la sorpresa Pagar cobrar al jugador actual
     * 
     */
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual,todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
    
    /**Sobreescribe el metodo toString
     * 
     */
    public String toString(){
        return "Pagar cobrar";
    }
}
